### PR TITLE
[CPDP-1202] - Your Capital Gains Tax on UK property return will be sa…

### PR DESCRIPTION
**Pull request by modivilioglu**
…ved for 30 days (saving return as draft)

### CODE REVIEW

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/ConfirmDraftReturnController.scala [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    request.sessionData.flatMap(_.journeyStatus) match {
+      case Some(FillingOutReturn(_, _, _, draftReturn)) => {
+        val draftReturnWithLastUpdated = draftReturn.copy(lastUpdatedDate = LocalDateUtils.today())
+
+        val response = returnsService.storeDraftReturn(draftReturnWithLastUpdated)
+
+        response.fold({ e =>
+          logger.error(
+            s"For cgt reference ${draftReturn.cgtReference.value}, got the following error ${e.value}")
+          errorHandler.errorResult()
```
Comment (mikail-khan) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/ConfirmDraftReturnController.scala]: could you add a test to exercise this branch

Comment (modivilioglu) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/ConfirmDraftReturnController.scala]: Done

File: conf/messages [hunk 2]
```diff
@@ -157,6 +157,9 @@ confirmEmail.verified.p1=We’ll use this address to contact you about your Capi
 confirmEmail.verified.p2=Your email has not been updated in other government services.
 confirmEmail.resent=Confirmation email sent:
 
+confirmDraftReturn.title=Your Capital Gains Tax on UK property return will be saved for 30 days
+confirmDraftReturn.message=After {0} all the details in your Capital Gains Tax on UK property return will be deleted. After that date, you'll need to start again. We will keep a return that is 'In progress' for 30 days from the day it was last saved. We will only save uploads for 7 days from the last day they were saved.
```
Comment (AndyHWChung) [inline conf/messages]: we use curly quotes in our messages like the ones here: https://smartquotesforsmartpeople.com/ You can copy and paste the left and right quotes from that page

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/confirmation_of_draft_return.scala.html [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+@message = @{messages("confirmDraftReturn.message", LocalDateUtils.govDisplayFormat(draftReturn.lastUpdatedDate.plusDays(29)))}
+
+@mainTemplate(title = title, userType = request.userType) {
+@backLinkComponent(returns.routes.TaskListController.taskList())
+
+@pageHeading(title)
+<div class="notice" style="margin: 30px 0;">
+    <i class="icon icon-important"></i>
+    <strong class="bold-small">
+        Your Capital Gains Tax on UK property return will be saved for 30 days
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/confirmation_of_draft_return.scala.html]: this sentence should come from `messages`

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/confirmation_of_draft_return.scala.html [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+@(draftReturn: DraftReturn)(implicit request: RequestWithSessionData[_], messages:Messages, viewConfig: ViewConfig)
+@title = @{messages("confirmDraftReturn.title")}
+@message = @{messages("confirmDraftReturn.message", LocalDateUtils.govDisplayFormat(draftReturn.lastUpdatedDate.plusDays(29)))}
+
+@mainTemplate(title = title, userType = request.userType) {
+@backLinkComponent(returns.routes.TaskListController.taskList())
+
+@pageHeading(title)
+<div class="notice" style="margin: 30px 0;">
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/confirmation_of_draft_return.scala.html]: @nubz  could you double check this template please? You'd know better than I would!

File: conf/messages (hunk 2 above)
Comment (AndyHWChung) [reply conf/messages]: also, shouldn't the sentences all be different `<p>`'s?

**Comment by AndyHWChung** [comment]
LGTM

**Pull request merged**